Unpin conda-build

jakirkham opened the pull request:
Reverts the pinning of `conda-build` to 2.1.4 as done in PR ( https://github.com/conda-forge/conda-forge-build-setup-feedstock/pull/61 ). As we recently packaged `conda-build` 2.1.8, which fixes the bugs experienced, we should be fine to drop this pinning. It also contains some useful fixes.

cc @conda-forge/core

jakirkham committed:
Revert "Pin conda-build to 2.1.4"

This reverts commit 451d0707fb74a846a7567268626d9fe8752d2693.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Ensure conda-build is installed just in case

Update of conda-build will fail if it is not present. So try installing
it no matter what so that we can guarantee it is present.

jakirkham committed:
Bump version to 4.4.4

jakirkham commented:
While we are waiting for CI, does anyone know of any reason for us not to unpin `conda-build`?
